src/zenml/integrations/aws/secrets_stores/aws_secrets_store.py

```diff
@@ -0,0 +1,1047 @@
+
+        Args:
+            secret_id: The ID of the secret to be updated.
+            secret_update: The update to be applied.
+
+        Returns:
+            The updated secret.
+
+        Raises:
+            EntityExistsError: If the update includes a change of name or
+                scope and a secret with the same name already exists in the
+                same scope.
+            RuntimeError: If the AWS Secrets Manager API returns an unexpected
+                error.
+        """
+        secret = self.get_secret(secret_id)
+        if secret_update.name is not None:
+            self._validate_aws_secret_name(secret_update.name)
+            secret.name = secret_update.name
+        if secret_update.scope is not None:
+            secret.scope = secret_update.scope
+        if secret_update.values is not None:
+            # Merge the existing values with the update values.
+            # The values that are set to `None` in the update are removed from
+            # the existing secret when we call `.secret_values` later.
+            secret.values.update(secret_update.values)
+
+        if secret_update.name is not None or secret_update.scope is not None:
+            # Check if a secret with the same name already exists in the same
+            # scope.
+            assert secret.user is not None
+            secret_exists, msg = self._check_secret_scope(
+                secret_name=secret.name,
+                scope=secret.scope,
+                workspace=secret.workspace.id,
+                user=secret.user.id,
+                exclude_secret_id=secret.id,
+            )
+            if secret_exists:
+                raise EntityExistsError(msg)
+
+        aws_secret_id = self._get_aws_secret_id(secret_id)
+        secret_value = json.dumps(secret.secret_values)
+
+        # Convert the ZenML secret metadata to AWS tags
+        metadata = self._get_secret_metadata_for_secret(secret)
+        tags = self._get_aws_secret_tags(metadata)
+
+        try:
+            # One call to update the secret values
+            self.client.put_secret_value(
+                SecretId=aws_secret_id,
+                SecretString=secret_value,
+            )
+            # Another call to update the tags
+            self.client.tag_resource(
+                SecretId=aws_secret_id,
+                Tags=tags,
+            )
+            # And another call to get the updated secret metadata which
+            # includes the created and updated timestamps.
+            describe_secret_response = self.client.describe_secret(
+                SecretId=aws_secret_id
+            )
+        except ClientError as e:
+            raise RuntimeError(f"Error updating secret: {e}")
+
+        logger.debug("Updated AWS secret: %s", aws_secret_id)
+
+        # We wait for the secret update to be reflected in the `list_secrets`
+        # API.
+        for _ in range(self.config.secret_list_refresh_timeout):
```

Comment:
Can we extract that into a separate method that also handles it in the `create_secret` case?

Reply:
done